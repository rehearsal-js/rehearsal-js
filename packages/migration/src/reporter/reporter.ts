import fs from 'fs';
import ts from 'typescript';
import winston from 'winston';
import json from './formatters/json';

export type ReportDiagnosticInfo = {
  file: string;
  code: number;
  message: string;
  start: number;
  length: number;
  nodeKind?: string;
  nodeText?: string;
};

export type Report = {
  version: string;
  diagnostics: ReportDiagnosticInfo[];
};

/**
 * Representation of diagnostic and migration report.
 */
export default class Reporter {
  readonly basePath: string;

  private report: Report;
  private logger?: winston.Logger;

  constructor(basePath = '', logger?: winston.Logger) {
    this.basePath = basePath;
    this.logger = logger?.child({ service: 'rehearsal-reporter' });
    this.report = {
      version: ts.version,
      diagnostics: [],
    };
  }

  getFileNames(): string[] {
    const fileNames: string[] = [];
    for (const diagnostic of this.report.diagnostics) {
      if (!fileNames.includes(diagnostic.file)) {
        fileNames.push(diagnostic.file);
      }
    }

    return fileNames;
  }

  getDiagnostics(fileName: string): ReportDiagnosticInfo[] {
    return this.report.diagnostics.filter((diagnostic) => diagnostic.file === fileName);
  }

  /**
   * Appends am information about provided diagnostic and related node to the report
   */
  addDiagnostic(diagnostic: ts.DiagnosticWithLocation, node?: ts.Node): void {
    this.report.diagnostics.push({
      file: diagnostic.file.fileName.replace(this.basePath, ''),
      code: diagnostic.code,
      message: ts.flattenDiagnosticMessageText(diagnostic.messageText, '. '),
      start: diagnostic.start,
      length: diagnostic.length,
      nodeKind: node ? ts.SyntaxKind[node.kind] : undefined,
      nodeText: node?.getText(),
    });
  }

  /**
   * Prints the current report using provided formatter (ex. json, pull-request etc.)
   */
  print(file: string, formatter: (report: Report) => string): string {
    const report = formatter(this.report);

    if (file) {
      fs.writeFileSync(file, report);
    }

    return report;
  }

  /**
   * Saves the current report information to the file in JSON format
   */
  save(file: string): void {
    this.print(file, json);
    this.logger?.info(`Report saved to: ${file}.`);
  }

  load(file: string): Reporter {
    if (!fs.existsSync(file)) {
      this.logger?.error(`Report file not found: ${file}.`);
    }

    this.logger?.info(`Report file found: ${file}.`);
    const content = fs.readFileSync(file, 'utf-8');
    const report = JSON.parse(content);

    if (!Reporter.isReport(report)) {
      this.logger?.error(`Report not loaded: wrong file format`);
    }

    this.report = report;
    this.logger?.info(`Report loaded from file.`);

    return this;
  }

  private static isReport(report: any): report is Report {
    return report && report.version !== undefined && report.diagnostics !== undefined;
  }
}
