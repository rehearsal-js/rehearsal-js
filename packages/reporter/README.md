# @rehearsal/reporter

Terminal and File reporter
