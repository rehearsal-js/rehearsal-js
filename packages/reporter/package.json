{
  "name": "@rehearsal/reporter",
  "version": "0.0.28",
  "description": "Rehearsal Reporter",
  "keywords": [
    "rehearsal",
    "reporter",
    "typescript"
  ],
  "repository": {
    "type": "git",
    "url": "https://github.com/rehearsal-js/rehearsal-js.git"
  },
  "license": "BSD-2-Clause",
  "main": "dist/src/index.js",
  "types": "dist/src/index.d.ts",
  "files": [
    "dist",
    "src"
  ],
  "scripts": {
    "build": "yarn tsc -b",
    "lint": "yarn npm-run-all lint:*",
    "lint:eslint": "yarn eslint --fix . --ext .ts",
    "lint:tsc-src": "yarn tsc --noEmit",
    "prepare": "yarn build",
    "test": "yarn nyc mocha"
  },
  "dependencies": {
    "winston": "^3.6.0"
  },
  "devDependencies": {
    "@types/chai": "^4.3.0",
    "@types/debug": "^4.1.7",
    "@types/eslint": "^8.4.1",
    "@types/mocha": "^9.1.0",
    "chai": "^4.3.6",
    "mocha": "^10.0.0",
    "nyc": "^15.1.0"
  },
  "peerDependencies": {
    "typescript": ">4.7"
  },
  "engines": {
    "node": ">=10.0.0"
  },
  "volta": {
    "node": "16.16.0",
    "yarn": "1.22.19"
  }
}
