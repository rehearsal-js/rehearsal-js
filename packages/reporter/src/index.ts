export { Reporter } from './reporter';
export { jsonFormatter } from './formatters/json-formatter';
export { mdFormatter } from './formatters/md-formatter';
export { sarifFormatter } from './formatters/sarif-formatter';
export { sonarqubeFormatter } from './formatters/sonarqube-formatter';
export { normalizeFilePath } from './normalize-paths';

export type { Location, Report, ReportItem, ReportSummary, ReportFormatter } from './types';
