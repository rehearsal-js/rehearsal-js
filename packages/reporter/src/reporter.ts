import { existsSync, readFileSync, writeFileSync } from 'fs';
import { DiagnosticCategory, flattenDiagnosticMessageText, SyntaxKind } from 'typescript';
import {
  type ProcessedFile,
  type Report,
  type ReportFormatter,
  type ReportItem,
  type Location,
} from './types';
import type { DiagnosticWithLocation, Node } from 'typescript';
import type { Logger } from 'winston';

type ReporterMeta = {
  projectName: string;
  basePath: string;
  commandName: string;
  tsVersion: string;
};

/**
 * Representation of diagnostic and migration report.
 */
export class Reporter {
  readonly basePath: string;

  public report: Report;
  private logger?: Logger;

  constructor(meta: ReporterMeta, logger?: Logger) {
    const { projectName, basePath, commandName, tsVersion } = meta;

    this.basePath = basePath;
    this.logger = logger?.child({ service: 'rehearsal-reporter' });

    this.report = {
      summary: {
        projectName: projectName,
        tsVersion: tsVersion,
        timestamp: new Date().toLocaleString('en-US', {
          year: 'numeric',
          month: 'numeric',
          day: 'numeric',
          hourCycle: 'h24',
          hour: '2-digit',
          minute: '2-digit',
          second: '2-digit',
        }),
        basePath: basePath,
        commandName: commandName,
      },
      items: [],
    };
  }

  getFileNames(): string[] {
    return [...new Set(this.report.items.map((item) => item.analysisTarget))];
  }

  getItemsByAnalysisTarget(fileName: string): ReportItem[] {
    return this.report.items.filter((item) => item.analysisTarget === fileName);
  }

  /**
   * Appends an element to the summary
   */
  addSummary(key: string, value: unknown): void {
    this.report.summary[key] = value;
  }

  /**
   * Appends am information about provided diagnostic and related node to the report
   */
  addItem(
    diagnostic: DiagnosticWithLocation,
    files: { [fileName: string]: ProcessedFile },
    fixed: boolean,
    node?: Node,
    triggeringLocation?: Location,
    hint = '',
    helpUrl = ''
  ): void {
    this.report.items.push({
      analysisTarget: diagnostic.file.fileName,
      files,
      errorCode: diagnostic.code,
      category: DiagnosticCategory[diagnostic.category],
      message: flattenDiagnosticMessageText(diagnostic.messageText, '. '),
      hint: hint,
      fixed,
      nodeKind: node ? SyntaxKind[node.kind] : undefined,
      nodeText: node?.getText(),
      helpUrl,
      //Original node that triggers the error; may not be the node where the codefix happens
      //Location in type ProcessedFile is where the codefix happens or the message is added
      //The two locations in most cases will be the same, but can be different
      nodeLocation: triggeringLocation || undefined,
    });
  }

  /**
   * Prints the current report using provided formatter (ex. json, pull-request etc.)
   */
  print(file: string, formatter: ReportFormatter): string {
    const report = formatter(this.report);

    if (file) {
      writeFileSync(file, report);
    }

    return report;
  }

  /**
   * Saves the current report information to the file in simple JSON format
   * to be able to load it later with 'load' function
   */
  save(file: string): void {
    this.print(file, (report: Report): string => JSON.stringify(report, null, 2));
    this.logger?.info(`Report saved to: ${file}.`);
  }

  /**
   * Loads the report exported by function 'save' from the file
   */
  load(file: string): Reporter {
    if (!existsSync(file)) {
      this.logger?.error(`Report file not found: ${file}.`);
    }

    this.logger?.info(`Report file found: ${file}.`);
    const content = readFileSync(file, 'utf-8');
    const report: Report = JSON.parse(content);

    if (!Reporter.isReport(report)) {
      this.logger?.error(`Report not loaded: wrong file format`);
      return this;
    }

    this.report = report as Report;
    this.logger?.info(`Report loaded from file.`);

    return this;
  }

  private static isReport(report: Report): report is Report {
    return report && report.summary !== undefined && report.items !== undefined;
  }
}
