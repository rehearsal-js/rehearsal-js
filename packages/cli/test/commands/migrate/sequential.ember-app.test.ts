import { resolve } from 'node:path';
import { readdirSync, promises as fs } from 'node:fs';
import { afterEach, beforeEach, afterAll, describe, expect, test, vi } from 'vitest';
import { createLogger, format, transports } from 'winston';
import { Report } from '@rehearsal/reporter';
import { Project } from 'fixturify-project';
import {
  getEmberAppProject,
  getEmberAppWithInRepoAddonProject,
  getEmberAppWithInRepoEngineProject,
  getEmber4AppProject,
} from '@rehearsal/test-support';
import {
  analyzeTask,
  depInstallTask,
  initTask,
  lintConfigTask,
  sequentialTask,
  tsConfigTask,
} from '../../../src/commands/migrate/tasks/index.js';

import {
  createMigrateOptions,
  createOutputStream,
  listrTaskRunner,
} from '../../test-helpers/index.js';

const logger = createLogger({
  transports: [new transports.Console({ format: format.cli() })],
});

const projects = {
  emberApp: getEmberAppProject(),
  emberAppWithInRepoAddon: getEmberAppWithInRepoAddonProject(),
  emberAppwithInRepoEngine: getEmberAppWithInRepoEngineProject(),
  ember4App: getEmber4AppProject(),
};

describe('Task: sequential - ember app', () => {
  for (const [name, originalProject] of Object.entries(projects)) {
    describe(name, () => {
      let project: Project;

      let output = '';
      let outputStream = createOutputStream();

      beforeEach(async () => {
        project = originalProject.clone();
        await project.write();

        outputStream = createOutputStream();

        vi.spyOn(console, 'info').mockImplementation((chunk: string) => {
          chunk = chunk.replace(new RegExp(project.baseDir, 'g'), '<tmp-path>');
          output += `${chunk}\n`;
          outputStream.push(`${chunk}\n`);
        });
        vi.spyOn(console, 'log').mockImplementation((chunk) => {
          output += `${chunk}\n`;
          outputStream.push(`${chunk}\n`);
        });
        vi.spyOn(console, 'error').mockImplementation((chunk) => {
          output += `${chunk}\n`;
          outputStream.push(`${chunk}\n`);
        });
      });

      afterEach(() => {
        output = '';
        vi.clearAllMocks();
        outputStream.destroy();
        project.dispose();
      });

      afterAll(() => {
        originalProject.dispose();
      });

      test('sequential run regen on the existing report, and run migrate on current base path and entrypoint', async () => {
        const options = createMigrateOptions(project.baseDir, { entrypoint: '', ci: true });
        const previousRuns = {
          previousFixedCount: 1,
          paths: [{ basePath: project.baseDir, entrypoint: 'foo.ts' }],
        };
        const tasks = [
          initTask(options),
          depInstallTask(options),
          tsConfigTask(options),
          lintConfigTask(options),
          analyzeTask(options),
          sequentialTask(options, logger, previousRuns),
        ];

        await listrTaskRunner(tasks);
        expect(output).toMatchSnapshot();

        const appPath = resolve(project.baseDir, 'app');
        const appFiles = readdirSync(appPath);
        expect(appFiles).toContain('app.ts');
        expect(appFiles).toContain('router.ts');
        expect(appFiles).toContain('index.html');

        const report = JSON.parse(
          await fs.readFile(resolve(project.baseDir, '.rehearsal', 'migrate-report.json'), 'utf-8')
        ) as Report;

        const { summary } = report;
        expect(summary?.length).toBe(2);
        expect(summary?.[0].basePath).toEqual(summary?.[1].basePath);
        expect(summary?.[0].entrypoint).toBe('foo.ts');
        expect(summary?.[1].entrypoint).toBe('');
      });
    });
  }
});
