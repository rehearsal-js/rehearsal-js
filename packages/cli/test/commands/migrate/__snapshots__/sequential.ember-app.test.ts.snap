// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Task: sequential - ember app > ember4App > sequential run regen on the existing report, and run migrate on current base path and entrypoint 1`] = `
"[STARTED] Initialize
[DATA] Setting up config for app-template
[SUCCESS] Initialize
[STARTED] Install dependencies
[DATA] Installing devDependencies: @types/node, @typescript-eslint/eslint-plugin, @typescript-eslint/parser, eslint, eslint-config-prettier, eslint-plugin-prettier, prettier, typescript
[SUCCESS] Install dependencies
[STARTED] Create tsconfig.json
[DATA] <tmp-path>/tsconfig.json already exists, ensuring strict mode is enabled
[TITLE] Update tsconfig.json
[SUCCESS] Update tsconfig.json
[STARTED] Create eslint config
[DATA] Create .eslintrc.js, extending Rehearsal default eslint-related config
[SUCCESS] Create eslint config
[STARTED] Analyzing Project
[DATA] Running migration on app-template
[SUCCESS] Analyzing Project
[STARTED] Regenerating past migrate report(s), generating current migrate report and merging all reports
[TITLE] Rescanning previously migrated files based on existing report:
[TITLE] 
[TITLE]   0 ts files have been scanned
[TITLE] 
[TITLE]   0 errors caught by rehearsal, and will be merged into the migration report
[TITLE] 
[TITLE]     
[DATA] git mv /app/app.js to /app/app.ts
[DATA] git mv /app/services/locale.js to /app/services/locale.ts
[DATA] git mv /app/components/salutation.js to /app/components/salutation.ts
[DATA] git mv /app/router.js to /app/router.ts
[DATA] git mv /tests/acceptance/index-test.js to /tests/acceptance/index-test.ts
[DATA] git mv /tests/test-helper.js to /tests/test-helper.ts
[DATA] git mv /tests/unit/services/locale-test.js to /tests/unit/services/locale-test.ts
[DATA] processing file: /app/app.ts
[DATA] processing file: /app/services/locale.ts
[DATA] processing file: /app/components/salutation.ts
[DATA] processing file: /app/router.ts
[DATA] processing file: /tests/acceptance/index-test.ts
[DATA] processing file: /tests/test-helper.ts
[DATA] processing file: /tests/unit/services/locale-test.ts
[TITLE] Migration Complete
[TITLE] 
[TITLE] 
[TITLE]   7 JS files converted to TS
[TITLE] 
[TITLE]   31 errors caught by rehearsal
[TITLE] 
[TITLE]   3 have been fixed by rehearsal
[TITLE] 
[TITLE]   28 errors need to be fixed manually
[TITLE] 
[TITLE]     -- 27 ts errors, marked by @ts-expect-error @rehearsal TODO
[TITLE] 
[TITLE]     -- 1 eslint errors, with details in the report
[TITLE] 
[SUCCESS] Migration Complete
[SUCCESS] 
[SUCCESS] 
[SUCCESS]   7 JS files converted to TS
[SUCCESS] 
[SUCCESS]   31 errors caught by rehearsal
[SUCCESS] 
[SUCCESS]   3 have been fixed by rehearsal
[SUCCESS] 
[SUCCESS]   28 errors need to be fixed manually
[SUCCESS] 
[SUCCESS]     -- 27 ts errors, marked by @ts-expect-error @rehearsal TODO
[SUCCESS] 
[SUCCESS]     -- 1 eslint errors, with details in the report
[SUCCESS] 
"
`;

exports[`Task: sequential - ember app > emberApp > sequential run regen on the existing report, and run migrate on current base path and entrypoint 1`] = `
"[STARTED] Initialize
[DATA] Setting up config for app-template
[SUCCESS] Initialize
[STARTED] Install dependencies
[DATA] Installing devDependencies: @types/node, @typescript-eslint/eslint-plugin, @typescript-eslint/parser, eslint, eslint-config-prettier, eslint-plugin-prettier, prettier, typescript
[SUCCESS] Install dependencies
[STARTED] Create tsconfig.json
[DATA] <tmp-path>/tsconfig.json already exists, ensuring strict mode is enabled
[TITLE] Update tsconfig.json
[SUCCESS] Update tsconfig.json
[STARTED] Create eslint config
[DATA] Create .eslintrc.js, extending Rehearsal default eslint-related config
[SUCCESS] Create eslint config
[STARTED] Analyzing Project
[DATA] Running migration on app-template
[SUCCESS] Analyzing Project
[STARTED] Regenerating past migrate report(s), generating current migrate report and merging all reports
[TITLE] Rescanning previously migrated files based on existing report:
[TITLE] 
[TITLE]   0 ts files have been scanned
[TITLE] 
[TITLE]   0 errors caught by rehearsal, and will be merged into the migration report
[TITLE] 
[TITLE]     
[DATA] git mv /app/app.js to /app/app.ts
[DATA] git mv /app/services/locale.js to /app/services/locale.ts
[DATA] git mv /app/components/salutation.js to /app/components/salutation.ts
[DATA] git mv /app/router.js to /app/router.ts
[DATA] git mv /tests/acceptance/index-test.js to /tests/acceptance/index-test.ts
[DATA] git mv /tests/test-helper.js to /tests/test-helper.ts
[DATA] git mv /tests/unit/services/locale-test.js to /tests/unit/services/locale-test.ts
[DATA] processing file: /app/app.ts
[DATA] processing file: /app/services/locale.ts
[DATA] processing file: /app/components/salutation.ts
[DATA] processing file: /app/router.ts
[DATA] processing file: /tests/acceptance/index-test.ts
[DATA] processing file: /tests/test-helper.ts
[DATA] processing file: /tests/unit/services/locale-test.ts
[TITLE] Migration Complete
[TITLE] 
[TITLE] 
[TITLE]   7 JS files converted to TS
[TITLE] 
[TITLE]   31 errors caught by rehearsal
[TITLE] 
[TITLE]   3 have been fixed by rehearsal
[TITLE] 
[TITLE]   28 errors need to be fixed manually
[TITLE] 
[TITLE]     -- 27 ts errors, marked by @ts-expect-error @rehearsal TODO
[TITLE] 
[TITLE]     -- 1 eslint errors, with details in the report
[TITLE] 
[SUCCESS] Migration Complete
[SUCCESS] 
[SUCCESS] 
[SUCCESS]   7 JS files converted to TS
[SUCCESS] 
[SUCCESS]   31 errors caught by rehearsal
[SUCCESS] 
[SUCCESS]   3 have been fixed by rehearsal
[SUCCESS] 
[SUCCESS]   28 errors need to be fixed manually
[SUCCESS] 
[SUCCESS]     -- 27 ts errors, marked by @ts-expect-error @rehearsal TODO
[SUCCESS] 
[SUCCESS]     -- 1 eslint errors, with details in the report
[SUCCESS] 
"
`;

exports[`Task: sequential - ember app > emberAppWithInRepoAddon > sequential run regen on the existing report, and run migrate on current base path and entrypoint 1`] = `
"[STARTED] Initialize
[DATA] Setting up config for app-template
[SUCCESS] Initialize
[STARTED] Install dependencies
[DATA] Installing devDependencies: @types/node, @typescript-eslint/eslint-plugin, @typescript-eslint/parser, eslint, eslint-config-prettier, eslint-plugin-prettier, prettier, typescript
[SUCCESS] Install dependencies
[STARTED] Create tsconfig.json
[DATA] <tmp-path>/tsconfig.json already exists, ensuring strict mode is enabled
[TITLE] Update tsconfig.json
[SUCCESS] Update tsconfig.json
[STARTED] Create eslint config
[DATA] Create .eslintrc.js, extending Rehearsal default eslint-related config
[SUCCESS] Create eslint config
[STARTED] Analyzing Project
[DATA] Running migration on app-template
[SUCCESS] Analyzing Project
[STARTED] Regenerating past migrate report(s), generating current migrate report and merging all reports
[TITLE] Rescanning previously migrated files based on existing report:
[TITLE] 
[TITLE]   0 ts files have been scanned
[TITLE] 
[TITLE]   0 errors caught by rehearsal, and will be merged into the migration report
[TITLE] 
[TITLE]     
[DATA] git mv /lib/some-addon/addon/components/greet.js to /lib/some-addon/addon/components/greet.ts
[DATA] git mv /app/app.js to /app/app.ts
[DATA] git mv /app/services/locale.js to /app/services/locale.ts
[DATA] git mv /app/components/salutation.js to /app/components/salutation.ts
[DATA] git mv /app/router.js to /app/router.ts
[DATA] git mv /tests/acceptance/index-test.js to /tests/acceptance/index-test.ts
[DATA] git mv /tests/test-helper.js to /tests/test-helper.ts
[DATA] git mv /tests/unit/services/locale-test.js to /tests/unit/services/locale-test.ts
[DATA] processing file: /lib/some-addon/addon/components/greet.ts
[DATA] processing file: /app/app.ts
[DATA] processing file: /app/services/locale.ts
[DATA] processing file: /app/components/salutation.ts
[DATA] processing file: /app/router.ts
[DATA] processing file: /tests/acceptance/index-test.ts
[DATA] processing file: /tests/test-helper.ts
[DATA] processing file: /tests/unit/services/locale-test.ts
[TITLE] Migration Complete
[TITLE] 
[TITLE] 
[TITLE]   8 JS files converted to TS
[TITLE] 
[TITLE]   32 errors caught by rehearsal
[TITLE] 
[TITLE]   3 have been fixed by rehearsal
[TITLE] 
[TITLE]   29 errors need to be fixed manually
[TITLE] 
[TITLE]     -- 28 ts errors, marked by @ts-expect-error @rehearsal TODO
[TITLE] 
[TITLE]     -- 1 eslint errors, with details in the report
[TITLE] 
[SUCCESS] Migration Complete
[SUCCESS] 
[SUCCESS] 
[SUCCESS]   8 JS files converted to TS
[SUCCESS] 
[SUCCESS]   32 errors caught by rehearsal
[SUCCESS] 
[SUCCESS]   3 have been fixed by rehearsal
[SUCCESS] 
[SUCCESS]   29 errors need to be fixed manually
[SUCCESS] 
[SUCCESS]     -- 28 ts errors, marked by @ts-expect-error @rehearsal TODO
[SUCCESS] 
[SUCCESS]     -- 1 eslint errors, with details in the report
[SUCCESS] 
"
`;

exports[`Task: sequential - ember app > emberAppwithInRepoEngine > sequential run regen on the existing report, and run migrate on current base path and entrypoint 1`] = `
"[STARTED] Initialize
[DATA] Setting up config for app-template
[SUCCESS] Initialize
[STARTED] Install dependencies
[DATA] Installing devDependencies: @types/node, @typescript-eslint/eslint-plugin, @typescript-eslint/parser, eslint, eslint-config-prettier, eslint-plugin-prettier, prettier, typescript
[SUCCESS] Install dependencies
[STARTED] Create tsconfig.json
[DATA] <tmp-path>/tsconfig.json already exists, ensuring strict mode is enabled
[TITLE] Update tsconfig.json
[SUCCESS] Update tsconfig.json
[STARTED] Create eslint config
[DATA] Create .eslintrc.js, extending Rehearsal default eslint-related config
[SUCCESS] Create eslint config
[STARTED] Analyzing Project
[DATA] Running migration on app-template
[SUCCESS] Analyzing Project
[STARTED] Regenerating past migrate report(s), generating current migrate report and merging all reports
[TITLE] Rescanning previously migrated files based on existing report:
[TITLE] 
[TITLE]   0 ts files have been scanned
[TITLE] 
[TITLE]   0 errors caught by rehearsal, and will be merged into the migration report
[TITLE] 
[TITLE]     
[DATA] git mv /lib/some-engine/addon/resolver.js to /lib/some-engine/addon/resolver.ts
[DATA] git mv /lib/some-engine/addon/engine.js to /lib/some-engine/addon/engine.ts
[DATA] git mv /lib/some-engine/addon/routes.js to /lib/some-engine/addon/routes.ts
[DATA] git mv /app/app.js to /app/app.ts
[DATA] git mv /app/services/locale.js to /app/services/locale.ts
[DATA] git mv /app/components/salutation.js to /app/components/salutation.ts
[DATA] git mv /app/router.js to /app/router.ts
[DATA] git mv /tests/acceptance/index-test.js to /tests/acceptance/index-test.ts
[DATA] git mv /tests/acceptance/some-engine-test.js to /tests/acceptance/some-engine-test.ts
[DATA] git mv /tests/test-helper.js to /tests/test-helper.ts
[DATA] git mv /tests/unit/services/locale-test.js to /tests/unit/services/locale-test.ts
[DATA] processing file: /lib/some-engine/addon/resolver.ts
[DATA] processing file: /lib/some-engine/addon/engine.ts
[DATA] processing file: /lib/some-engine/addon/routes.ts
[DATA] processing file: /app/app.ts
[DATA] processing file: /app/services/locale.ts
[DATA] processing file: /app/components/salutation.ts
[DATA] processing file: /app/router.ts
[DATA] processing file: /tests/acceptance/index-test.ts
[DATA] processing file: /tests/acceptance/some-engine-test.ts
[DATA] processing file: /tests/test-helper.ts
[DATA] processing file: /tests/unit/services/locale-test.ts
[TITLE] Migration Complete
[TITLE] 
[TITLE] 
[TITLE]   11 JS files converted to TS
[TITLE] 
[TITLE]   42 errors caught by rehearsal
[TITLE] 
[TITLE]   3 have been fixed by rehearsal
[TITLE] 
[TITLE]   39 errors need to be fixed manually
[TITLE] 
[TITLE]     -- 39 ts errors, marked by @ts-expect-error @rehearsal TODO
[TITLE] 
[TITLE]     -- 0 eslint errors, with details in the report
[TITLE] 
[SUCCESS] Migration Complete
[SUCCESS] 
[SUCCESS] 
[SUCCESS]   11 JS files converted to TS
[SUCCESS] 
[SUCCESS]   42 errors caught by rehearsal
[SUCCESS] 
[SUCCESS]   3 have been fixed by rehearsal
[SUCCESS] 
[SUCCESS]   39 errors need to be fixed manually
[SUCCESS] 
[SUCCESS]     -- 39 ts errors, marked by @ts-expect-error @rehearsal TODO
[SUCCESS] 
[SUCCESS]     -- 0 eslint errors, with details in the report
[SUCCESS] 
"
`;
